src/scripts/main.js

```diff
@@ -1,3 +1,361 @@
+          return b.children[0].innerText.localeCompare(a.children[0].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Position':
+        sort = listTable.sort((a, b) => {
+          return b.children[1].innerText.localeCompare(a.children[1].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Office':
+        sort = listTable.sort((a, b) => {
+          return b.children[2].innerText.localeCompare(a.children[2].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Age':
+        sort = listTable.sort((a, b) => {
+          return b.children[3].innerText - a.children[3].innerText;
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Salary':
+        sort = listTable.sort((a, b) => {
+          return convertNum(b.children[4].innerText)
+            - convertNum(a.children[4].innerText);
+        });
+
+        table.append(...sort);
+    }
+  } else {
+    nameTitle = e.target.innerText;
+
+    switch (e.target.innerText) {
+      case 'Name':
+        sort = listTable.sort((a, b) => {
+          return a.children[0].innerText.localeCompare(b.children[0].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Position':
+        sort = listTable.sort((a, b) => {
+          return a.children[1].innerText.localeCompare(b.children[1].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Office':
+        sort = listTable.sort((a, b) => {
+          return a.children[2].innerText.localeCompare(b.children[2].innerText);
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Age':
+        sort = listTable.sort((a, b) => {
+          return a.children[3].innerText - b.children[3].innerText;
+        });
+
+        table.append(...sort);
+        break;
+
+      case 'Salary':
+        sort = listTable.sort((a, b) => {
+          return convertNum(a.children[4].innerText)
+            - convertNum(b.children[4].innerText);
+        });
+
+        table.append(...sort);
+    }
+  }
+});
+
+table.addEventListener('click', (e) => {
+  [...document.querySelectorAll('tr')].slice(1, -1).forEach((item) => {
+    item.classList = '';
+  });
+
+  e.path[1].className = 'active';
+});
+
+const massageResultTrue = Promise.resolve('new employer created');
+const massageResultFalse = Promise.reject(Error('some wrong, write correct'));
+
+document.querySelector('button').addEventListener('click', e => {
+  const div = document.createElement('div');
+
+  div.dataset.qa = 'notification';
+
+  e.preventDefault();
+
+  const dataInput = document.querySelectorAll('input');
+  const selectSity = document.querySelector('select');
+  const optionName = document.querySelectorAll('option');
+
+  let controlForm = true;
+
+  if (dataInput[0].value.length < 4 || +dataInput[0].value) {
+    controlForm = false;
```

Comment:
instead of using `controlForm` you can show a unique message for each field: resolve your promise (actually it could be a simple function) and write a message here